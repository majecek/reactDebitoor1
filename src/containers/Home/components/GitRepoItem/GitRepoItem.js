import React from 'react'
import PropTypes from 'prop-types'
import { TableRow, TableRowColumn } from 'material-ui/Table'
import { Link } from 'react-router'

const GitRepoItem = ({ repo }) => (
  <TableRow selectable={false}>
    <TableRowColumn >{repo.name}</TableRowColumn>
    <TableRowColumn >{repo.stargazers_count}</TableRowColumn>
    <TableRowColumn >{repo.stargazers_count}</TableRowColumn>
    <TableRowColumn >{repo.open_issues_count}</TableRowColumn>
    <TableRowColumn ><Link to={'/repo/' + repo.full_name}>Current PRs</Link></TableRowColumn>
  </TableRow>
)

GitRepoItem.propTypes = {
  repo: PropTypes.object.isRequired
}

export default GitRepoItem