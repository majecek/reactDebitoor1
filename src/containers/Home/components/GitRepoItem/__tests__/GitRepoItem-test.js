import React from 'react'
import renderer from 'react-test-renderer'
import GitRepoItem from '../GitRepoItem'
import baseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
const theme = getMuiTheme(baseTheme)

describe('GitRepoItem', () => {
  const repo = {
    name: 'foo',
    stargazers_count: 100,
    open_issues_count: 10,
    full_name: 'bar'
  }
  const result = renderer.create(
    <MuiThemeProvider muiTheme={theme}>
      <GitRepoItem repo={repo} />
    </MuiThemeProvider>).toJSON()

  it('should render full html structure correctly', () => {
    expect(result).toMatchSnapshot()
  })
})