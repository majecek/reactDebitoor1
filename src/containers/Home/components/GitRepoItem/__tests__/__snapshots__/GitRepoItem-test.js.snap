exports[`GitRepoItem should render full html structure correctly 1`] = `
<tr
  className={undefined}
  style={
    Object {
      "borderBottom": "1px solid #e0e0e0",
      "color": "rgba(0, 0, 0, 0.87)",
      "height": 48,
      "muiPrepared": true,
    }
  }>
  <td
    className={undefined}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "backgroundColor": "inherit",
        "fontSize": 13,
        "height": 48,
        "muiPrepared": true,
        "overflow": "hidden",
        "paddingLeft": 24,
        "paddingRight": 24,
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
      }
    }>
    foo
  </td>
  <td
    className={undefined}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "backgroundColor": "inherit",
        "fontSize": 13,
        "height": 48,
        "muiPrepared": true,
        "overflow": "hidden",
        "paddingLeft": 24,
        "paddingRight": 24,
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
      }
    }>
    100
  </td>
  <td
    className={undefined}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "backgroundColor": "inherit",
        "fontSize": 13,
        "height": 48,
        "muiPrepared": true,
        "overflow": "hidden",
        "paddingLeft": 24,
        "paddingRight": 24,
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
      }
    }>
    100
  </td>
  <td
    className={undefined}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "backgroundColor": "inherit",
        "fontSize": 13,
        "height": 48,
        "muiPrepared": true,
        "overflow": "hidden",
        "paddingLeft": 24,
        "paddingRight": 24,
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
      }
    }>
    10
  </td>
  <td
    className={undefined}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "backgroundColor": "inherit",
        "fontSize": 13,
        "height": 48,
        "muiPrepared": true,
        "overflow": "hidden",
        "paddingLeft": 24,
        "paddingRight": 24,
        "textAlign": "left",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
      }
    }>
    <a
      onClick={[Function]}
      style={Object {}}>
      Current PRs
    </a>
  </td>
</tr>
`;
